resolver/src/main/java/io/netty/resolver/AddressResolverGroup.java

```diff
@@ -59,29 +59,31 @@ protected AddressResolverGroup() { }
             throw new IllegalStateException("executor not accepting a task");
         }
 
-        AddressResolver<T> r;
-        synchronized (resolvers) {
-            r = resolvers.get(executor);
-            if (r == null) {
-                final AddressResolver<T> newResolver;
-                try {
-                    newResolver = newResolver(executor);
-                } catch (Exception e) {
-                    throw new IllegalStateException("failed to create a new resolver", e);
-                }
+        AddressResolver<T> r = resolvers.get(executor);
```

Comment:
@carryxyh it's not safe to do this since the map used here is `IdentityHashMap` which is not threadsafe. If this is a bottleneck we would need to change the map impl.

Reply:
Check the code and doc: 

```
 * <p><strong>Note that this implementation is not synchronized.</strong>
 * If multiple threads access an identity hash map concurrently, and at
 * least one of the threads modifies the map structurally, it <i>must</i>
 * be synchronized externally.  (A structural modification is any operation
 * that adds or deletes one or more mappings; merely changing the value
 * associated with a key that an instance already contains is not a
 * structural modification.)  This is typically accomplished by
 * synchronizing on some object that naturally encapsulates the map.
```

There is no doubt that you are right.. Thx for your review. Close this.